source/particle/generator/reference_cell.cc

```diff
@@ -160,8 +157,9 @@ namespace aspect
             {
               prm.enter_subsection("Reference cell");
               {
-                for (int i=0; i<dim; i++)
-                  number_of_particles[i] = (unsigned int) (prm.get_double ("Number of tracers per cell per direction"));
+                number_of_particles = Utilities::possibly_extend_from_1_to_N (Utilities::string_to_double(Utilities::split_string_list(prm.get("Number of tracers per cell per direction"))),
```

Comment:
Here you can introduce a temporary `std::vector<double>` to store the intermediate result and then copy and cast it to `unsigned int` to avoid the casts during the actual particle generation. This could also make the lines somewhat shorter.